stac_geoparquet/arrow/_to_arrow.py

```diff
@@ -1,48 +1,33 @@
 """Convert STAC data into Arrow tables"""
 
-import json
-from datetime import datetime
 from pathlib import Path
 from typing import (
     Any,
     Dict,
-    Generator,
     Iterable,
     Iterator,
-    List,
     Optional,
-    Sequence,
     Union,
 )
 
-import ciso8601
-import numpy as np
 import pyarrow as pa
-import pyarrow.compute as pc
-import shapely
-import shapely.geometry
 
-from stac_geoparquet.arrow._schema.models import InferredSchema
-from stac_geoparquet.arrow._crs import WGS84_CRS_JSON
-from stac_geoparquet.arrow._util import stac_items_to_arrow
-
-
-def _chunks(
-    lst: Sequence[Dict[str, Any]], n: int
-) -> Generator[Sequence[Dict[str, Any]], None, None]:
-    """Yield successive n-sized chunks from lst."""
-    for i in range(0, len(lst), n):
-        yield lst[i : i + n]
+from stac_geoparquet.json_reader import read_json
+from stac_geoparquet.arrow._util import (
+    stac_items_to_arrow,
+    batched_iter,
+    update_batch_schema,
+)
 
 
-def parse_stac_items_to_arrow(
-    items: Sequence[Dict[str, Any]],
+def parse_stac_items_to_batches(
+    items: Iterable[Dict[str, Any]],
     *,
     chunk_size: int = 8192,
-    schema: Optional[Union[pa.Schema, InferredSchema]] = None,
```

Comment:
Why remove the `InferredSchema` support here? I had intended `InferredSchema` to act the same as a `pa.Schema`, but with stronger typing (e.g. `InferredSchema` is just a newtype around a schema)